back/app/controllers/web_api/v1/participation_contexts_controller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class WebApi::V1::ParticipationContextsController < ::ApplicationController
+  before_action :set_context_no_clash, only: %i[survey_results submission_count delete_inputs]
+
+  def survey_results
+    results = SurveyResultsGeneratorService.new(@context).generate_results
+    render json: results
+  end
+
+  def submission_count
+    count = SurveyResultsGeneratorService.new(@context).generate_submission_count
+    render json: count
+  end
+
+  def delete_inputs
+    sidefx.before_delete_inputs @context, current_user
+    ActiveRecord::Base.transaction do
+      @context.ideas.each(&:destroy!)
+    end
+    sidefx.before_delete_inputs @context, current_user
```

Comment:
```suggestion
    sidefx.after_delete_inputs @context, current_user
```
In principle the acceptance tests should verify that the side effect is created. I know it is not done usually, but it would be better, otherwise we do not describe that the side effects are triggered.

Should`after_delete_inputs` be done if the transaction was rolled back for some reason?

Reply:
Okay, I will add checks for calling after_delete_inputs.

I cannot think of a scenario where the deletion of ideas would fail, which is why I couldn't come up with a test for that case. I would consider anything that would rollback the transaction as a bug. Therefore, I expect that Rails will throw an exception if something goes wrong, undoing all previous deletions, and skipping the after_delete_inputs to return a 500. Sentry would then list the issue so we can quickly solve it.

Reply:
> I expect that Rails will throw an exception if something goes wrong, undoing all previous deletions

Indeed. So no action needed for that aspect of my comment :-)

Reply:
I don't see an easy way to check that `after_delete_inputs` is called through an acceptance test. Shall I write controller tests instead?

Reply:
Would this work?
```
expect_any_instance_of(SideFxPhaseService).to receive(:after_delete_inputs)
```
That does not verify the arguments of calling `after_delete_inputs`, but it verifies that it calls it. If it calls it with the wrong methods, I expect issues elsewhere.